Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -1732,8 +1731,17 @@ void CClientPed::InternalSetHealth(float fHealth)
         // Recheck we have a ped, ReCreateModel might destroy it
         if (m_pPlayerPed)
         {
+            // update dead state for peds
+            if (fHealth > 0 && IsDead())
+            {
+                m_bDead = false;
+            }
+
             // Set the new health
             m_pPlayerPed->SetHealth(fHealth);
+
+            // Recover from dead state to bring the ped back to life
+            m_pTaskManager->RemoveTask(TASK_SIMPLE_DEAD);
```

Comment:
When two or more players are streamed in, this PR crashes the targeted player via `setElementHealth` (crash occurs with both client and server-side variant of the function). Other clients that were streamed in will crash upon unload.

I am trying to get more information about the crash right now.

So far:
 - Seems to be a `m_pTaskManager` related crash. You've added `m_pTaskManager->RemoveTask(TASK_SIMPLE_DEAD);` which is likely the source of the crash.

```
Version = 1.5.6-custom.0.000
Time = Sat Feb 23 16:43:08 2019
Module = F:\Program Files\Steam\steamapps\common\Grand Theft Auto San Andreas\gta_sa.exe
Code = 0xC0000005
Offset = 0x00281B46

EAX=00000000  EBX=00000000  ECX=0000000B  EDX=21704F14  ESI=21704F14
EDI=000000DA  EBP=0177F348  ESP=0177F31C  EIP=00681B46  FLG=00010202
CS=0023   DS=002B  SS=002B  ES=002B   FS=0053  GS=002B
```
Used artifact: https://ci.appveyor.com/project/multitheftauto/mtasa-blue/builds/22594951/artifacts

Reply:
I'm not sure, but maybe we should use ```TASK_PRIORITY_PRIMARY``` instead of ```TASK_SIMPLE_DEAD``` and maybe it will work.  I don't know why this crashes game. i can't do investigate :/